src/Types/DateTimeType.php

```diff
@@ -44,11 +47,26 @@ public function convertToDatabaseValue($value, AbstractPlatform $platform)
             return $value;
         }
 
+        if ($value instanceof DateTimeImmutable) {
+            Deprecation::triggerIfCalledFromOutside(
+                'doctrine/dbal',
+                'https://github.com/doctrine/dbal/pull/6017',
+                'Passing an instance of %s is deprecated, use %s::%s() instead.',
+                get_class($value),
```

Comment:
Not sure if this is relevant, but the use of `$value::class` would so the same thing ans was used on many occasions in Doctrine repos. But that's nitpicking.

Reply:
We are supporting PHP 7.4. See https://3v4l.org/CuEBQ#v7.4.0.